aea/runtime.py

```diff
@@ -62,14 +69,88 @@ def __init__(
         :return: None
         """
         self._agent: "Agent" = agent
-        self._loop = ensure_loop(
+        self._loop: AbstractEventLoop = ensure_loop(
             loop
         )  # TODO: decide who constructs loop: agent, runtime, multiplexer.
         self._state: AsyncState = AsyncState(RuntimeStates.stopped, RuntimeStates)
         self._state.add_callback(self._log_runtime_state)
         self._was_started = False
 
+        self._multiplexer: Multiplexer = self._get_multiplexer_instance()
+        self._task_manager = TaskManager()
+        self._decision_maker: Optional[DecisionMaker] = None
+
+        self.main_loop: BaseAgentLoop = self._get_main_loop_instance(
+            agent._loop_mode or self.DEFAULT_RUN_LOOP
```

Comment:
loop mode should be a property on agent; no private access

Reply:
although; why not pass it to the constructer of the runtime from the agent?

Reply:
nice point, actually  agent should not care about how it's running
done